site/docs/advanced/flood.md

```diff
@@ -5,29 +5,119 @@ next: ./transformers.md
 
-Telegram restricts how many messages your bot can send per second, confer the [Bot FAQ](https://core.telegram.org/bots/faq#my-bot-is-hitting-limits-how-do-i-avoid-this).
-You should always make sure to stay below these limits, otherwise your bot gets rate limited.
-If you ignore these errors, your bot may eventually be banned.
+Telegram restricts how many messages your bot can send per second.
+This means that any API request you perform could error with status code 429 (Too Many Requests) and a `retry_after` header as specified [here](https://core.telegram.org/bots/api#responseparameters).
+This can happen anytime.
 
-## The Simple Solution
+There is only one correct way to handle such errors.
 
-:::warning Not a Real Solution
-This section solves your problem short-term, but if you are building a bot that should actually scale well, read the [next subsection](#the-real-solution-recommended) instead.
-:::
+1. Wait for the specified number of seconds.
+2. Retry the request.
 
-There is a very simple solution to hitting rate limits: if an API request fails due to a rate limit, just wait the time Telegram tells you to wait, and repeat the request.
+Fortunetaly, there is a [plugin](../plugins/auto-retry.md) for that.
 
-If you want to do this, you can use the [super simple `auto-retry` plugin](../plugins/auto-retry.md).
-It is an [API transformer function](./transformers.md) that does exactly that.
+That plugin is [very simple](https://github.com/grammyjs/auto-retry/blob/main/src/index.ts).
+It literally just sleeps and retries.
+However, using it has a major implication: **any request can be slow**.
+This means that when you run your bot on webhooks, [you technically have to use a queue](../guide/deployment-types.md#ending-webhook-requests-in-time) no matter what you do, or else you need to configure the auto-retry plugin in a way that it never takes a lot of time---but then your bot may skip some requests.
 
-However, if the traffic to your bot increases rapidly, e.g. when it is added to a large group, it may run into a lot of rate limiting errors before the traffic spike settles.
-This could lead to a ban.
-Moreover, as requests might be tried several times, your server will consume more RAM and bandwidth than necessary.
-Instead of fixing the problem after the fact, it is much better to enqueue all API requests and only send them at the permitted speed:
+## What the Exact Limits Are
 
-## The Real Solution (recommended)
+They are unspecified.
 
-grammY provides you with the [throttler plugin](../plugins/transformer-throttler.md) that automatically makes your bot respect all rate limits by enqueuing the outgoing requests of your bot in a message queue.
-This plugin is just as simple to set up but does a much better job at flood control.
-There isn't really any good reason to use [auto-retry](../plugins/auto-retry.md) over the [throttler plugin](../plugins/transformer-throttler.md).
-In some cases it may make sense to use both.
+Deal with it.
```

Comment:
```suggestion
You need to deal with them.
```

Reply:
That's not what I meant. I am referring to the meme.